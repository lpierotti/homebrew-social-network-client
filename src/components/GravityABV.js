import React from 'react'

function GravityABV(props) {

	const handleChange = (event) => {
		props.changeGravity(event.target.name, event.target.value)
	}
	
	return (
		<div>
			<input name='og' type='number' step="0.01" onChange={handleChange} placeholder='OG'/>
			<input name='fg' type='number' step="0.01" onChange={handleChange} placeholder='FG'/>
			<p>{props.data.abv}%</p>
		</div>
	)
}

export default GravityABV