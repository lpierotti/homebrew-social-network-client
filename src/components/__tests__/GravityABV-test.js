import React from 'react';
import { shallow, mount, render } from 'enzyme';
import GravityABV from '../GravityABV';
import renderer from 'react-test-renderer';

describe('GravityABV component', () => {

	it('renders correctly', () => {
		const tree = renderer.create(
			<GravityABV changeGravity={() => {}} data={{}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	})
})