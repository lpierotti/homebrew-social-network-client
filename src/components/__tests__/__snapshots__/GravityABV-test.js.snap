// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GravityABV component renders correctly 1`] = `
<div
  className="equal width fields"
>
  <div
    className="field"
  >
    <div
      className="ui input"
    >
      <input
        disabled={undefined}
        name="og"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        placeholder="OG"
        required={undefined}
        step="0.01"
        tabIndex={undefined}
        type="number"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui input"
    >
      <input
        disabled={undefined}
        name="fg"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        placeholder="FG"
        required={undefined}
        step="0.01"
        tabIndex={undefined}
        type="number"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui input"
    >
      <input
        disabled={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        placeholder="ABV"
        required={undefined}
        tabIndex={undefined}
        type="text"
        value="undefined%"
      />
    </div>
  </div>
</div>
`;
